README.md

```diff
@@ -1,10 +1,145 @@
 # Godot engine build containers
 
-This repository contains the Dockerfiles for the official Godot engine builds. These containers should help you build Godot for all platforms supported on any machine that can run Docker containers.
+This repository contains the Dockerfiles for the official Godot engine builds.
+These containers should help you build Godot for all platforms supported on
+any machine that can run Docker containers.
 
-## Building
+The in-container build scripts are in a separate repository:
+https://github.com/godotengine/godot-build-scripts
 
-There is a 'build.sh' script included to build the containers themselves.
 
-The in-container build scripts are in a separate repository: https://github.com/godotengine/godot-build-scripts
+## Introduction
+
+These scripts build a number of containers which are then used to build final
+Godot tools, templates and server packages for several platforms.
+
+Once these containers are built, they can be used to compile different Godot
+versions without the need of recreating them.
+
+The `upload.sh` file is meant to be used by Godot Release Team and is not
+documented here.
+
+
+## Requirements
+
+These containers have been tested under Fedora 33 and Ubuntu 18.04 (others may work too).
+
+The tool used to build and manage the containers is `podman`.
+
+See the Host OS section below for further information on how to setup your host OS before start.
+
+
+## Usage
+
+The 'build.sh' script included is used to build the containers themselves.
+
+Run the command using:
+
+    ./build.sh 3.2 mono-6.12.0.114 2020-02
```

Comment:
The `2020-02` part is not necessary, there are two forms with either 2 or 4 arguments:
```
  echo "Usage: $0 <godot branch> <mono version> [<mono branch> <mono commit hash>]"
  echo
  echo "Examples: $0 3.1 mono-5.18.1.3"
  echo "        $0 master mono-6.6.0.160 2019-08 bef1e6335812d32f8eab648c0228fc624b9f8357"
```

The form with two arguments relies on a `mono/mono` tag (here `mono-6.12.0.114`) so the branch name is not needed. The form with four arguments is to specify both the branch and the commit (and then the second argument is not a tag but just the name to put in the image version).